File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/friend/operation/BlacklistCancelResponse.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.friend.operation;

import com.zfoo.protocol.IPacket;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2019-11-13 18:41
 */
public class BlacklistCancelResponse implements IPacket {

    public static final transient short PROTOCOL_ID = 15111;

    private long targetId;

    public static BlacklistCancelResponse valueOf(long targetId) {
        var response = new BlacklistCancelResponse();
        response.targetId = targetId;
        return response;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

    public long getTargetId() {
        return targetId;
    }

    public void setTargetId(long targetId) {
        this.targetId = targetId;
    }
}

File: zapp-gateway/jsProtocol/friend/operation/BlacklistCancelResponse.js
// @author jaysunxiao
// @version 1.0
// @since 2019-11-13 18:41
const BlacklistCancelResponse = function(targetId) {
    this.targetId = targetId; // long
};

BlacklistCancelResponse.prototype.protocolId = function() {
    return 15111;
};

BlacklistCancelResponse.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writeLong(packet.targetId);
};

BlacklistCancelResponse.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new BlacklistCancelResponse();
    const result0 = buffer.readLong();
    packet.targetId = result0;
    return packet;
};

export default BlacklistCancelResponse;
